unixfs/tar/writer.go

```diff
@@ -21,79 +21,100 @@ import (
 // TODO: does this need to be configurable?
 var DefaultBufSize = 1048576
 
-func DagArchive(ctx cxt.Context, nd *mdag.Node, name string, dag mdag.DAGService, compression int) (io.Reader, error) {
+// DagArchive is equivalent to `ipfs getdag $hash | maybe_tar | maybe_gzip`
+func DagArchive(ctx cxt.Context, nd *mdag.Node, name string, dag mdag.DAGService, useTar bool, compression int) (io.Reader, error) {
 
 	_, filename := path.Split(name)
 
 	// need to connect a writer to a reader
 	piper, pipew := io.Pipe()
 
-	// use a buffered writer to parallelize task
-	bufw := bufio.NewWriterSize(pipew, DefaultBufSize)
```

Comment:
the bufio.Writer is here so that we do not need to proceed in lockstep. i think the bufio.Writer will make it so that a worker goroutine can be fetching dag parts and writing them, and a different goroutine can be reading them. the bufio.Writer also has a reasonably big buffer (~3-5 blocks)